### DISCUSSION TITLE
Typescript support with subscriptions seems wonky

### DISCUSSION THREAD

**Discussion by BigBallard** [vuejs/pinia#1492] [Help and Questions] [answered]
I am trying to use Pinia in a project and can't seem to resolve this typescript linting issue with the `$subscription`.
![error](https://user-images.githubusercontent.com/13224109/181655183-047324d0-e9cc-4251-92e3-1d6f524b99e8.png)
I essentially want to get the specific value for the mutation or at least check the payload for the value changed and do something with it but this is blocking me from doing that simple thing and I want to use Pinia since it is now the default.

**Comment by posva** [comment, marked as answer]
Only `$patch()` generates a payload: https://pinia.vuejs.org/core-concepts/state.html#subscribing-to-the-state

**Comment by BigBallard** [comment]
Ya I know, that doesn't answer the question though. The image is showing an TS typing problem that I am trying to investigate.

**Reply by posva** [reply]
Did you read the link? 😅

![Screenshot 2022-07-29 at 18 28 21](https://user-images.githubusercontent.com/664177/181803599-f942d1a5-b06b-40e3-b91f-71f453842b52.png)

So:

```ts
if (mutation.type === 'patch object') {
  mutation.payload // valid!
}
```

**Reply by ZiuChen** [reply]
help me alot! ❤
